server/src/pages/shot/view.js

```diff
@@ -340,19 +339,21 @@ class Body extends React.Component {
     const linkTextShort = shot.urlDisplay;
 
     const timeDiff = <TimeDiff date={shot.createdDate} />;
-    let expiresDiff = null;
-    if (this.props.isOwner) {
-      expiresDiff = <span className="expire-widget">
-      <ExpireWidget
-        expireTime={this.props.expireTime}
-        onChanging={this.onChangingExpire.bind(this)}
-        onSaveExpire={this.onSaveExpire.bind(this)} />
-      </span>;
-    }
 
+    let favoriteShotButton = null;
     let trashOrFlagButton;
     let editButton;
     const highlight = this.state.highlightEditButton ? <div className="edit-highlight" onClick={ this.onClickEdit.bind(this) } onMouseOver={ this.onMouseOverHighlight.bind(this) } onMouseOut={ this.onMouseOutHighlight.bind(this) }></div> : null;
+
+    if (this.props.isFxaAuthenticated) {
+      const activeFavClass = this.props.expireTime ? "" : "is-fav";
+      favoriteShotButton = <div className="fav-wrapper"><button
```

Comment:
This `div` is here because brightness filter on the icon image affects the background color, which make the hover and active background inconsistent with the other icons.